platform/default/default_file_source.cpp

```diff
@@ -79,9 +79,9 @@ class DefaultFileSource::Impl {
     }
 
     void getRegionStatus(int64_t regionID, std::function<void (std::exception_ptr, optional<OfflineRegionStatus>)> callback) {
-        try {
-            callback({}, getDownload(regionID).getStatus());
-        } catch (...) {
+        if (auto download = getDownload(regionID)) {
+            callback({}, download->getStatus());
+        } else {
             callback(std::current_exception(), {});
```

Comment:
`std::current_exception()` will always return an empty result here, since it's no longer inside a catch block.